src/term.rs

```diff
@@ -110,6 +113,7 @@ impl<VS: VersionSet> Term<VS> {
     pub(crate) fn is_disjoint(&self, other: &Self) -> bool {
         match (self, other) {
             (Self::Positive(r1), Self::Positive(r2)) => r1.is_disjoint(r2),
+            // Unselected package is allowed in both terms.
             (Self::Negative(r1), Self::Negative(r2)) => r1 == &VS::empty() && r2 == &VS::empty(),
```

Comment:
Should the result always be `false` here, since both terms contains the unselected package ?

Reply:
I don’t remember exactly when this was written, but my gut feeling is that this was created mainly to be used in multiple other places where the contents (semantic) of this function was already pre-existing. Some kind of refactoring, where an appropriate need was needed, and `is_disjoint` seemed like the best match.

The exact refactor happened here: https://github.com/pubgrub-rs/pubgrub/commit/0d80e9d40d13ce1df96ad8da84a263d28ae6e6ba

Reply:
Otherwise, I agree with you that two negative terms are never technically disjoint, so maybe there is an issue? Would be surprising considering the amount of tests around the project, but not impossible.

Reply:
We often handle the empty case specially. For example it gets filtered out of incompatibilities. So I would not be surprised if in the end-to-end  tests every time we call `is_disjoint` one of the arguments has been already determined to be non-empty.

On the other hand, I'm surprised that were passing `is_disjoint_through_intersection`.   Perhaps our strategy is not generating the empty set often enough? Yup, looks like we generate ranges with `1..10`  segments which does not gen the empty set.

Reply:
The `is_disjoint_through_intersection` test fails if we restrict the generated range to `0..1` in the proptest strategy.

Reply:
🎉 nice find!

Let's modify this line to always return false with a comment about why, and modify the generation to `0..10` so that we don't regress in the future.

Reply:
Agreed. Thanks for the finding!